runtime/interpreter/account.go

```diff
@@ -86,6 +86,15 @@ func NewAuthAccountValue(
 		sema.AuthAccountStoragePathsField: func(inter *Interpreter, getLocationRange func() LocationRange) Value {
 			return inter.storageAccountPaths(address, getLocationRange)
 		},
+		sema.AuthAccountForEachPublicField: func(inter *Interpreter, _ func() LocationRange) Value {
+			return inter.iterOverStorageDomain(address, common.PathDomainPublic, sema.PublicPathType)
```

Comment:
The `Value` here is the `*HostValueFunction`. I'm not sure what a `HostValueFunction` is but should it be invoked here?

Reply:
No. These define a method on the account object. They are invoked when the method is called, not when it is defined.